### ISSUE TITLE
Seeing unnatural high amount of calls to S3ListObject when running BoltDB Shipper

### ISSUE THREAD

**Issue by andersosthus** [grafana/loki#3258]
**Describe the bug**
After activating detailed metrics on our S3 Loki Bucket, we saw that we had a very large volume of calls to S3ListObject, about 3M per day, which seemed way too high. I talked a bit with @owen-d about it, and after some digging around I found the following:
- We have 62 tables in S3 stored with BoltDB
- We run 5 queriers

Each querier seem to be doing multiple calls per minute to `sync` tables, as evident by the following log query:
`{namespace="kube-loki"} |~"(?i)todownload" !="query_type"`:
![image](https://user-images.githubusercontent.com/184104/106203538-d2e60280-61bb-11eb-8178-2eaac7085ff6.png)

I also looked at the metric `loki_boltdb_shipper_tables_sync_operation_total`, and summed by `job`, it looks like this for the last 24 hours (`sum by (job) (loki_boltdb_shipper_tables_sync_operation_total)`):
![image](https://user-images.githubusercontent.com/184104/106205129-4b4dc300-61be-11eb-9b61-26be68305911.png)

Basically, `querier` has about 80000 and `ruler` about 17000 for the last 24 hours.

According to the code, this gets incremented here https://github.com/grafana/loki/blob/cd723c0479fc8bbdd1a01b6cee0e129503f58d08/pkg/storage/stores/shipper/downloads/table_manager.go#L205
A few lines after, it does a `table.Sync` call for each table in `tableManager`, which ends up here as a call to S3 ListObjects https://github.com/grafana/loki/blob/6d442673b7279eef4b7aa88d0af0ceb72bba7fe9/pkg/storage/stores/shipper/downloads/table.go#L401

**Expected behavior**
Fewer calls to S3ListObjects

**Environment:**
 - Infrastructure: Kubernetes
 - Deployment tool: Manual
 - Loki 2.1

Tagging @owen-d and @sandeepsukhani as requested by Owen :)

**Comment by sandeepsukhani** [comment]
3M calls per day sounds too much for the size of your cluster. We run 20 queriers and 2 rulers in one of our Loki clusters with 30 days retention and it is doing ~2M List calls per day overall. I know it is still a lot and I have an idea to improve it but haven't had a chance to work on it.
Can you please share your config to see why it is so high for your?

**Comment by andersosthus** [comment]
I think I might have figured it out. I went through the config, and looked at the [online docs](https://grafana.com/docs/loki/v2.1.0/configuration/) trying to figure it out, and then I noticed under `storage_config.boltdb_shipper` there is a setting named `resync_interval`, which, afaik, is undocumented. It's not listed (neither is any of the `storage_config.boltdb_shipper`) on the official Configuration page, and under Operations-Storage-Single Store, it's mentioned in the text below, but not in the main example.

Anyways, my `boltdb-shipper` config was based on https://github.com/grafana/loki/blob/f6fd6ae7da0d929788250a1273efc531ae25ec65/docs/sources/configuration/examples.md, which sets the `resync_interval` to `5s`. I've changed it to `1m` now, and already I see that the `loki_boltdb_shipper_tables_sync_operation_total` is reporting a lot less table sync operations.

Maybe the `boltdb_shipper` parts of the config should be added to the main Configuration section, and also the examples should probably not have `resync_interval` set at all since it defaults to `15m` according to the docs?

**Comment by sandeepsukhani** [comment]
Yeah, I was aiming to check `resync_interval`. Sorry about all the inconvenience that you faced.
I will fix the docs and example config.
Also, the default value for `resync_interval` is `5m`, if you can point me to where you saw it being `15m`, I can fix that too. You can increase it if you want. We internally run it with the default value.

**Comment by andersosthus** [comment]
No worries :) Glad to have figured it out.

I got the implication that it was 15m from this page: https://grafana.com/docs/loki/v2.1.0/operations/storage/boltdb-shipper/#queriers, second paragraph where it talks about the `resync_interval`.

**Comment by iamparamvs** [comment]
Hi @sandeepsukhani  
Shouldn't [this](https://github.com/grafana/loki/blob/f6fd6ae7da0d929788250a1273efc531ae25ec65/docs/sources/configuration/examples.md#almost-zero-dependencies-setup) documentation also be changed for resync_interval from 5s to 5m ? I think this is the reason we have kept resync_interval to 5s while using it with S3.
